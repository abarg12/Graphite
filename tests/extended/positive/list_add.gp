list l = [];
list l2 = ["false", 1, 2.2, "three", false];

int i = array_get(l2, 1);
printf(i);
array_add(l2, 1, "hi");
string s = array_get(l2, 1);
printf(s);


array_add(l, 0, "0ind");
string s2 = array_get(l, 0);
printf(s2);


void add_to_empty(list l1) {
    array_add(l1, 0, "newnodejustdropped");
    string s = array_get(l1, 0);
    printf(s);
    return;
}

{
    add_to_empty(l);
}
