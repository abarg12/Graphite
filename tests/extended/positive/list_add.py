list l = [];
list l2 = [false, 1, 2.2, "three"];

# array_add(l, 0, "whats");
int j = 2;
string s = "sillystring";
s = array_get(l2, 3);
float f = array_get(l2, 2);
printf(s);

# float f = array_get(l2, 2);
# printf(f);
# array_add(l2, 2, "new");

# int i;
# for (i = 0; i < 5; i = i+1) {
#     f = array_get(l2, i);
#     printf(f);
# }

