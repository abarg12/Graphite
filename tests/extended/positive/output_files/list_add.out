1
hi
0ind
newnodejustdropped
